src/lzw.ts

```diff
@@ -0,0 +1,131 @@
+import { IOBuffer } from 'iobuffer';
+
+const CLEAR_CODE = 256;
+const EOI_CODE = 257;
+// 0-255 from the table + 256 for clear code + 257 for end of information code.
+const TABLE_START = 258;
+const MIN_BIT_LENGTH = 9;
+
+const stringTable: number[][] = [];
+for (let i = 0; i < 256; i++) {
+  stringTable.push([i]);
+}
+// Fill the table with dummy data.
+// Elements at indices > 257 will be replaced during decompression.
+const dummyString = [0];
+for (let i = 256; i < 4096; i++) {
+  stringTable.push(dummyString);
+}
```

Comment:
This table is going to be always kept in memory, with whatever data remains after the last decompression.
I think it's fine since the length is not too big. This should increase the performance and decrease the pressure on the GC (otherwise a new table must be created for each strip of data in the image)